src/sagemaker_xgboost_container/data_utils.py

```diff
@@ -485,6 +485,52 @@ def get_recordio_protobuf_dmatrix(path, is_pipe=False):
         raise exc.UserError("Failed to load recordio-protobuf data with exception:\n{}".format(e))
 
 
+def _get_pipe_mode_files_path(data_path):
```

Comment:
```suggestion
def _get_pipe_mode_files_path(data_path: Union[List[str], str]) -> List[str]:
```

Same below. Let's incrementally make our functions less ambiguous with type hints.